scripts/apps/authoring/metadata/metadata.ts

```diff
@@ -364,13 +364,17 @@ function MetaDropdownDirective($filter) {
 
                 if (item) {
                     if (scope.cv) {
+                        // if there is cv as well as field, store cv._id as scheme
+                        // so that it can be differentiated from another cv inside same parent field(subject).
+                        // ex: subject:[{name: "a", qcode: "a", scheme: "new-cv"}]
                         item.scheme = scope.cv._id;
                         o[scope.field] = scope.item[scope.field].filter((v) => v.scheme !== scope.cv._id);
                         o[scope.field].push(item);
                     } else {
                         o[scope.field] = scope.key ? item[scope.key] : [item];
                     }
                 } else if (scope.cv) {
+                    // if there is cv as well as field don't set o[scope.field] to null.
```

Comment:
What is `o`? I know it's not you who added, but renaming to a more meaningful name would help.

> don't set o[scope.field] to null.

What does setting to null do?

Reply:
some of the fields like `subject` are set as `nullable: false` in the backend.

Reply:
Add it to the comment in the code :)